consensus/core/src/ancestor.rs

```diff
@@ -0,0 +1,381 @@
+                    );
+                    self.context
+                        .metrics
+                        .node_metrics
+                        .ancestor_state_change_by_authority
+                        .with_label_values(&[block_hostname, "exclude"])
+                        .inc();
+                }
+            }
+            // Check conditions to switch back to INCLUDE state
+            AncestorState::Exclude(_) => {
+                // It should not be possible for the scores to get over the threshold
+                // until the node is back in the INCLUDE state, but adding just in case.
+                if propagation_score > low_score_threshold
+                    || authority_low_quorum_round >= (network_low_quorum_round)
+                {
+                    ancestor_info.state = AncestorState::Include;
+                    ancestor_info.set_lock(
+                        self.propagation_score_update_count + Self::STATE_LOCK_SCORE_UPDATES,
+                    );
+                    info!(
+                        "Authority {authority_id} moved to INCLUDE state with {propagation_score} > threshold of {low_score_threshold} or {authority_low_quorum_round} >= {network_low_quorum_round} and locked for {:?} score updates.",
+                        Self::STATE_LOCK_SCORE_UPDATES
+                    );
+                    self.context
+                        .metrics
+                        .node_metrics
+                        .ancestor_state_change_by_authority
+                        .with_label_values(&[block_hostname, "include"])
+                        .inc();
+                }
+            }
+        }
+
+        // If any updates were made to state ensure they are persisted.
+        self.state_map[authority_id] = ancestor_info;
+    }
+
+    /// Calculate the network's quorum round from authorities by inclusion stake
+    /// threshold, where quorum round is the highest round a block has been seen
+    /// by a percentage (inclusion threshold) of authorities.
+    fn calculate_network_low_quorum_round(&self) -> u32 {
+        let committee = &self.context.committee;
+        let inclusion_stake_threshold = self.get_inclusion_stake_threshold();
+        let mut low_quorum_rounds_with_stake = self
+            .quorum_round_per_authority
+            .iter()
+            .zip(committee.authorities())
+            .map(|((low, _high), (_, authority))| (*low, authority.stake))
+            .collect::<Vec<_>>();
+        low_quorum_rounds_with_stake.sort();
+
+        let mut total_stake = 0;
+        let mut network_low_quorum_round = 0;
+
+        for (round, stake) in low_quorum_rounds_with_stake.iter().rev() {
+            let reached_quorum_before = total_stake >= inclusion_stake_threshold;
+            total_stake += stake;
+            if !reached_quorum_before && total_stake >= inclusion_stake_threshold {
+                network_low_quorum_round = *round;
+                break;
+            }
```

Comment:
Probably this could be reduced to:

```
            total_stake += stake;
            if total_stake >= inclusion_stake_threshold {
                network_low_quorum_round = *round;
                break;
            }
```